### ISSUE TITLE
[HOLD for payment 2024-12-30] [Awaiting payment][$250] Mark down - The sent message containing two > does not display double quote mark down

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45154]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.6-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Go to chat
3. Type >> hello
4. Note that the live mark down displays the nested quote
5. Send the message

## Expected Result:
The sent message containing two > will display double quote mark down

## Actual Result:
The sent message containing two > does not display double quote mark down

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/dccd694c-d8c6-4022-a99f-1721185b16bf

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021856771063028418492</li>
        <li>Upwork Job ID: 1856771063028418492</li>
        <li>Last Price Increase: 2024-11-13</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@pecanoro  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by pecanoro** [comment]
Confirmed and posted in the PR, we probably need to assign the people to fix it!

**Comment by pecanoro** [comment]
Oh, but maybe we don't need to hold on this as it only happens when you do >> which it's an edge case

**Comment by mountiny** [comment]
Yeah lets not hold on this, we have to get to higher version of the markdown library and this does not seem that big issue

Please feel free to reassign to me, happy to handle this with swm or community upstream

**Comment by mountiny** [comment]
cc @Skalakid @BrtqKr @BartoszGrajdek @tomekzaw

**Comment by BartoszGrajdek** [comment]
I think @BrtqKr will be the best person to help you here since he was the one working on block quotes 🙌🏻

**Comment by BrtqKr** [comment]
Blockquote requires a space following it. If there are two blockquotes, they still need to be separated with a space. This was meant to allow empty lines to be treated as a part of the blockquote and prevent some of the issues with treating '>' as a blockquote unintentionally. I think this should go the other way - the input shouldn't show that this would result in a blockquote.

**Comment by pecanoro** [comment]
@BrtqKr Let me know if I am following correctly, the problem here is that the composer should not show the preview of the markdown, right?

**Comment by pecanoro** [comment]
@BrtqKr Any updates on when we will have this one fixed?

**Comment by BrtqKr** [comment]
@pecanoro @lanitochka17 @JmillsExpensify There have been a lot of issues related to the behavior of blockquotes, describing different, sometimes conflicting results as desired. We've been discussing it internally and we think we should take a different approach in this case. Can we please:

1. Pass [this document ](https://docs.google.com/document/d/18YEgGIZSLuGrOsfVJkQ7Q1i1wUi23KGQGy0ZWlQYlKg/edit?usp=sharing) further inside the Expensify team. 
2. Make the decisions regarding cases described in the document.
3. Create a single ticket to adjust those behaviors on both sides - the input and the parser. And redirect all blockquotes-related reports to that place

We think that we're lacking a single source of truth with an appropriate description for the parser and the general approach to fixing those things seems a bit chaotic and blindly adjusting regexes without a broad context might not be the best workflow.

**Comment by pecanoro** [comment]
@JmillsExpensify Since it's product design related, could you handle this? https://github.com/Expensify/App/issues/45154#issuecomment-2252238031

**Comment by ChavdaSachin** [comment]
Checkout alternate method I suggested in [this proposal](https://github.com/Expensify/App/issues/47951#issuecomment-2308868522) It solves this issue as well and we will gain similar behavior for composer and report message. 
Issue -https://github.com/Expensify/App/issues/47951#issue-2484028828

**Comment by JmillsExpensify** [comment]
Aligning in a doc sounds good to me.

**Comment by JmillsExpensify** [comment]
I'll take a look this week.

**Comment by ChavdaSachin** [comment]
@JmillsExpensify I have proposals ready for both cases - 
1. To mimic composer behavior for, ie. add support for multiple '>' without space in between.
2. To align composer preview with comment markdown, ie. '>' without space in between would not show quote markdown in composer.
If you want external contributor to work on this issue then let me know once you finalize expected behavior.

**Comment by JmillsExpensify** [comment]
Let me post on this internally and make sure we have consensus.

**Comment by JmillsExpensify** [comment]
Whoops, sorry forgot to help push this one forward. I'll circle back on it this week instead.

**Comment by JmillsExpensify** [comment]
We're de-prioritizing these kinds of issues based on recent changes in the roadmap, so I'm just going to close this one out.

**Comment by zanyrenney** [comment]
Hey @JmillsExpensify as a result of the conversation in slack, I am reopening this one. 

I was going to just keep it in the new issue but upon reviewing this issue, seeing as there has already been a contributor slated to work on it and a fair amount of conversation about it in this issue, it seemed like a duplication and lack of context to start this issue afresh. Happy to manage it as the BZ assignee though if you like!

**Comment by zanyrenney** [comment]
Feel free to assign me and swap yourself out if you prefer :)

**Comment by zanyrenney** [comment]
Just noting that this was reported again [here](https://github.com/Expensify/App/issues/52318) -

**Comment by Skalakid** [comment]
Hi, I'm Michał from SWM, an expert agency, and I will investigate this issue. More info in [the Slack thread](https://swmansion.slack.com/archives/C049HHMV9SM/p1731574717153529?thread_ts=1731423572.626789&cid=C049HHMV9SM)

**Comment by Skalakid** [comment]
Hello, Here is the [draft PR](https://github.com/Expensify/expensify-common/pull/820) with the fix. I changed the blockquote parsing logic and now I'm working on changing html to markdown conversion algorithm

**Comment by Skalakid** [comment]
Hello, today I continued fixing HTML to markdown parsing logic. Tomorrow I will clean up the code and push the changes to the PR so we can start merging it

**Comment by pecanoro** [comment]
@Skalakid Any recent updates?

**Comment by Skalakid** [comment]
Hi, yes, I'm adding last touches to my [PR](https://github.com/Expensify/expensify-common/pull/820) and will handle it to the review today

**Comment by pecanoro** [comment]
@Pujan92 Do you want to review the PR? Since you were assigned here originally as C+

**Comment by pecanoro** [comment]
@Pujan92 It is not letting me assigned you so you probably need to comment in the PR

**Comment by Pujan92** [comment]
@pecanoro Yes, I will review it tomorrow.

**Comment by Skalakid** [comment]
[PR](https://github.com/Expensify/expensify-common/pull/820) is waiting for the C+ review

**Comment by Skalakid** [comment]
The PR has been merged. We will bump the Expensify-common together with the next Live Markdown bump since some new breaking changes were introduced there

**Comment by Skalakid** [comment]
Waiting for this [PR](https://github.com/Expensify/App/pull/53627/files) to be merged, since it bumps the Live Markdown and expensify-common libraries

**Comment by Skalakid** [comment]
Oh, now I found out that blockquotes can be nested infinitely after my exepnsify-common changes... Creating follow up PR

**Comment by pecanoro** [comment]
@JmillsExpensify I think the only thing left here is pay @Pujan92 for the review

**Comment by JmillsExpensify** [comment]
Payment summary above is correct. @Pujan92 do you mind completing the BZ checklist?

**Comment by Pujan92** [comment]
## Regression Test Proposal

1. Go to any chat
2. Type message **>> Hello** OR any valid blockquote format
3. Note that the live markdown displays the nested quote
4. Send the message
5. Verify the sent message should also display the markdown message correctly(nested blockquote)

**Comment by garrettmknight** [comment]
$250 approved for @Pujan92
